joomla/joomla-cms
# [4.0] Defer all scripts by default, second try. So much speed, wow!

### Summary of Changes

The patch make all JavaScripts deferred by default, even inline scripts (which now has a type "module" by default).

This is another try (first was #22460 by @laoneo)
Since we dropped IE support the things become much easier.



And when we said "module" then should come @dgrammatiko and say "module!" :wink: 


### Testing Instructions

Apply patch, and make sure all works as before. 
Enable debug mode and check all still works as before.

Please test in all (supported) browser you have.

### Documentation Changes Required

I think it may need to add a note to https://docs.joomla.org/Potential_backward_compatibility_issues_in_Joomla_4
This introduce a little B.C. changes for people who have used inline `<script>` or `<script type="text/javascript">`.
And can be easily fixed by changing to `<script type="module">`.

Example old code: `<script>jQuery(document).ready(function(){ .... })</script>`
The new version: `<script type="module">jQuery(document).ready(function(){ .... })</script>`

However there no B.C. for people who have used `$doc->addScriptDeclaration($code)`;
